test/kubernetes/tools.go

```diff
@@ -271,18 +271,41 @@ func CoreDNSPodIPs() ([]string, error) {
 	return ips, nil
 }
 
-// HasCoreDNSRestarted verifies if any of the CoreDNS containers has restarted.
-func HasCoreDNSRestarted() (bool, error) {
-	restartCount, err := Kubectl("-n kube-system get pods -l k8s-app=kube-dns -ojsonpath='{.items[*].status.containerStatuses[0].restartCount}'")
+// HasResourceRestarted verifies if any of the specified containers in the kube-system namespace has restarted.
+func HasResourceRestarted(label string) (bool, error) {
+	// magic number
+	wait := 5
+
+	for {
+		restartCount, err := Kubectl(fmt.Sprintf("-n kube-system get pods -l %s -ojsonpath='{.items[*].status.containerStatuses[0].restartCount}'", label))
+		if err != nil {
+			return false, err
+		}
+		if !strings.Contains(restartCount, "0") {
```

Comment:
Technically here restarting 10 times would be seen as not restarting, because it contains a 0. 
I don't think a container will realistically restart 10 times in 1 second, but if one did, then this could conclude that there were no restarts after 5 seconds of checking (0,10,20,30,40,50).  

It would be technically more correct to split the result on the white space, and check each string == "0".

Reply:
Makes sense! Fixed